app/javascript/components/async-credentials/async-credentials.jsx

```diff
@@ -70,9 +68,10 @@ const AsyncCredentials = ({
   };
 
   return (
-    <PasswordContext.Provider value={{ name, edit }}>
+    <Fragment>
       {formOptions.renderForm(fields.map(field => ({
         ...field,
+        ...(field.component === 'password-field' ? { parent: name, edit } : undefined),
```

Comment:
What is the parent and how is it used?

I feel like that is getting dangerously close to `$scope.parent`

Reply:
This is the argument formerly known as `secretKey`, you were passing it as a context. I just renamed it to something more meaningful, but I can switch it back.

Reply:
I am not talking about the name. I am worried about the `parent`.

Reply:
Yeah, is there anything different from what you implemented before where it was named differently but it did exactly the same, just through a context instead of a prop... :confused: